### DISCUSSION TITLE
2023 Superset Community Survey - Question Suggestions

### DISCUSSION THREAD

**Discussion by rusackas** [apache/superset#24968] [Ideas]
Hi everyone,

As proposed on the Superset dev mailing list, this discussion will serve the purpose of gathering questions to add to the annual Superset Community Survey. The goal is to attempt to reach ALL Superset users/maintainers, and determine as much as we can about the product's global use. We encourage everyone to respond, as **the best way to help open source is to simply let people know you use it** (and how/where/why).

### Types of questions to suggest
Questions should be generally applicable toward the Superset community. That is to say we're looking for aggregatable stats and demographics, like "which version are you running?" or "which dev setup are you using" or "have you contributed code?" but not "can we have Feature X?" or "when will Bug Y be fixed?

### How to use this discussion
Questions should be added on separate response blocks below, and discussions regarding each submitted questions should be added as inline comments below it. 

### Question selection/voting
If you like or dislike a questions please feel free to give them a 👍 or 👎 - these will be considered as advisory votes when volunteers from the PMC compile the final survey.

This thread will remain locked until this proposal has sufficient consensus to proceed. The thread will be locked again at the end of the submission window (TBD).

**Comment by mistercrunch** [comment]
Which area(s) would you consider contributing to if you had more time, or if it was easier to get started?
- documentation
- advocacy: blog/social/promotion
- frontend
- backend
- code quality
- security / secops
- devex/devops

**Reply by rudokemper** [reply]
* frontend
* backend

**Reply by webobite** [reply]
- Documentation 
- Frontend 
- Back end
- security / secops

**Comment by mistercrunch** [comment]
From a UX standpoint, which area of Superset do you think is more important to invest into and uplevel (pick 2)?
- dashboard consumption
- dashboard creation
- data exploration
- content discovery (tags, folder, global search, homepage)
- SQL Lab
- Alerts & Reports
- chart quality
- theming
- ... (feel free to add more options!)

**Reply by jayakrishnankk** [reply]
embedded superset

**Reply by rudokemper** [reply]
* internationalization and translation
* theming

**Reply by j-meher** [reply]
- copilot for Superset

**Reply by webobite** [reply]
- Edit chart controls: metrics layer control in edit chart of superset ( to be more specific)

**Comment by mistercrunch** [comment]
Which database engines do you use in your organization (pick many)?

**Reply by rusackas** [reply]
We need a comprehensive list, but yes. This is also a good chance to ask people to sign up for [this](https://docs.google.com/spreadsheets/d/1U1qxiLvOX0kBTUGME1AHHi6Ywel6ECF8xk_Qy-V9R8c/edit#gid=0) if it's not overloading things too much.

**Comment by mistercrunch** [comment]
In your organization, how many users use Superset monthly (MAU)?
- just me!
- just a few people
- more than a dozen
- 50+
- 100+
- 250+
- 500+
- 1000+!

**Reply by sfirke** [reply]
Would we want to break this out into authors vs. readers?

**Comment by mistercrunch** [comment]
Would your organization be open to be public about the fact that they use Superset? (the idea here is that it would be great if we could feature a list of logo/orgs of companies who use Superset in the docs and/or GitHub readme)

**Reply by satokonak** [reply]
The decision is usually up to the organization's PR team?

**Comment by michael-s-molina** [comment]
We could have some questions to measure overall sentiment using a 1-5 emoji scale.

![image](https://github.com/apache/superset/assets/70410625/e4d3f33b-9aa2-461d-93ab-5ef3215a24de)

How would you rate Superset for each of the following topics:
- Stability
- Security
- Code quality
- Community support
- Performance
- UI/UX
- Developer experience
- Documentation
- Ease of installation/configuration
...

**Comment by rusackas** [comment]
By what means do you install and/or deploy Superset?
- [ ] Docker/docker-compose
- [ ] Python (pip/virtualenv)
- [ ] Helm charts
- [ ] I use a hosted provider (e.g. Preset)
- [ ] Other

**Comment by rusackas** [comment]
Via @mistercrunch (moving this to be its own item)
> If we were to build more collaborative features into Superset, what integration would matter most to your organization?
>
> * Slack!
> * Microsoft Teams
> * [whatever else people use]

**Reply by rusackas** [reply]
Via @jayakrishnankk (moving this to a comment on the new thread)

> * google chat

**Reply by mdeshmu** [reply]
Generic Webhook

**Comment by pandinug** [comment]
In your organisation, Superset is primarily used for:

- Self-service (ad-hoc) analytics
- Guided analytics

**Reply by satokonak** [reply]
What does "guided analytics" mean?

**Comment by pandinug** [comment]
In your organisation, Superset is primarily used to get insight on the level of:

- Strategic performance (organisation wide longer term goals)
- Tactical performance (department-based goals)
- Operational performance (high-detail + granularity on your business processes)

**Comment by rusackas** [comment]
The NPS question:
"How likely is it that you would recommend Superset to a friend or colleague?" (score from 1-10)

**Reply by rusackas** [reply]
We might just skip this one, unless people feel strongly about it.

**Comment by satokonak** [comment]


**Reply by rusackas** [reply]
Is there a way to get an aggregatable answer for this, you think? I'm worried about being unable to reconcile a million typed-up bug reports 😅

**Reply by satokonak** [reply]
sg lets ignore

**Comment by satokonak** [comment]
What AI or generative AI features do you want to see in Superset?

**Reply by rusackas** [reply]
Should we give them a list of choices here, or leave this open-ended? Giving options might measure impact of "soft proposals" but we'd have to be clear we're not making any promises.

**Reply by satokonak** [reply]
I was thinking of open-ended question to generate ideas; I can make this an optional question

**Comment by satokonak** [comment]
Which groups in your organization use Superset (select all that apply)?
- [ ] Data / Data Platform teams
- [ ] Product & Engineering 
- [ ] Sales, Marketing, & Customer Success
- [ ] Finance, Legal, and HR
- [ ] Other

**Comment by satokonak** [comment]
Which BI tools does your company / organization use? Select all that apply. 

- [ ] Superset
- [ ] Tableau
- [ ] PowerBI
- [ ] Qlik
- [ ] Looker
- [ ] Other [name]

**Reply by sfirke** [reply]
Prior to Superset, what BI tool(s) did your company use:

- [ ] Tableau
- [ ] PowerBI
- [ ] Qlik
- [ ] Looker
- [ ] Metabase
- [ ] ...
- [ ] Other [name]
- [ ] None, Superset is our organization's first BI tool

**Comment by yousoph** [comment]
Which Superset features are most valuable to you?

**Reply by rusackas** [reply]
What would be the answers here... multiple choice in terms of areas (Chart Building, Dashboarding, SQL Querying) or do we want to get more granular here?

**Reply by rusackas** [reply]
We can also ask the inverse, with the same answer options: Which features do you find confusing or difficult to use?

**Reply by yousoph** [reply]
I was thinking open ended might give us more detailed answers in case the multiple choice options are too broad for what a respondent has in mind, but open to making it multiple choice with an open text Other option perhaps

**Comment by yousoph** [comment]
What is one thing that you feel is missing from Superset?

**Comment by sfirke** [comment]
In what way(s) do you interact with Superset?

- [ ] Consumer of charts/dashboards
- [ ] Author of charts/dashboards
- [ ] Administrator of Superset deployment
- [ ] Software developer of Superset's code

**Comment by rusackas** [comment]
How did you first hear about Superset?
- [ ] Search
- [ ] My company/org
- [ ] Blogs / articles
- [ ] Apache or other Apache Projects
- [ ] Other (specify)

**Comment by rusackas** [comment]
Some branching in this one, but to get a sense of developer engagement:

Have you considered contributing to the Superset codebase?
- [ ] Yes
   - Have you opened a PR?
     - [ ] Yes
       - How difficult was the process?
         - [ ] 1-10 Likert scale
       - How likely are you to do it again?
         - [ ] 1-10 Likert scale
    - [ ] No
      - What might've prevented you from opening a PR?
        - [ ] Unsure if it would help
        - [ ] The codebase is too complex
        - [ ] I don't have the time
        - [ ] I'm not allowed to
        - [ ] Other (specify)
- [ ] No
  - Is there anything we could do differently that would make you interested in contribute code to Superset? (open answer, optional)

**Reply by michael-s-molina** [reply]
@rusackas It would be good if we can use the same scale for the questions. I like 1-5 as defined [here](https://github.com/apache/superset/discussions/24968#discussioncomment-6744740) but I'm not opposed to use 1-10. The important part is that they should be consistent.

**Reply by rusackas** [reply]
Agreed on consistency. If we do an NPS score, that's a 10 point scale as a standard. Would you allow that one inconsistency if the rest were a 5 point scale?

**Reply by michael-s-molina** [reply]
No problem. I'm fine with using the 1-10 score for all questions too.

**Comment by rusackas** [comment]
How easy is it to find the information you need in the documentation?
- [ ] 1-10 Leikert score

**Comment by rusackas** [comment]
What version of Superset are you using/running?
- [ ] {list of recent-ish versions here}
- [ ] I run off master
- [ ] I use a hosted service, e.g. Preset
- [ ] I run off a forked and modified version
- [ ] Other/I'm not sure

**Reply by satokonak** [reply]
To make this mutually exclusive, I'll remove the version option (that can be a separate question if important)

**Comment by rusackas** [comment]
Are you a developer running Superset on your local machine?
- [ ] Y
  - What was the purpose of this?
    - [ ] To customize Superset (development purposes)
    - [ ] To test the product
    - [ ] I use Superset in isolation
  - How do you install superset?
    - [ ] Docker-compose
    - [ ] Python virtualenv
    - [ ] Makefile
    - [ ] Kubernetes/Helm
    - [ ] PyPi download
    - [ ] Other (specify)
- [ ] N

**Reply by satokonak** [reply]
There is a separate question for the installation one

**Comment by rusackas** [comment]
Are you reading/enjoying the Superset Community Newsletter?
- [ ] Heck yeah!
  - Any feedback/suggestions?
    - [ ] Open answer textarea
- [ ] There's a newsletter?
- [ ] I've unsubscribed

**Comment by rusackas** [comment]
How would you rate the effectiveness of communication within the community?
- [ ] 1-10 scale

**Comment by rusackas** [comment]
What improvements can be made to community communication channels?
- [ ] open-ended textarea

**Comment by rusackas** [comment]
What is your primary role as it pertains to Superset Use??
- [ ] Software engineer 
- [ ] Data analyst
- [ ] Student/Academic
- [ ] Manager
- [ ] Stakeholder (operations, finances, etc)
- [ ] Other (specify)

I'm sure there are different options for this... please add some comments :D

**Reply by satokonak** [reply]
There's a similar question!

**Comment by rusackas** [comment]
How would you rate your Superset expertise as an end-user?
- [ ] 1-10 Likert scale

**Reply by satokonak** [reply]
There's a similar question!

**Comment by rusackas** [comment]
What are your preferred means to keep up to date on a project and its community? (select all that apply)
- [ ] Slack
- [ ] Discord
- [ ] Email
- [ ] Twitter
- [ ] Facebook
- [ ] Other social (specify)
- [ ] Other (specify)

**Comment by rusackas** [comment]
What database(s) are you connecting Superset to?
- [ ] {insert 1000 checkboxes here}
- [ ] Other(s) - please specify

**Reply by satokonak** [reply]
We have a similar question above!

**Comment by rusackas** [comment]
What organization do you work for (optional)
- If they answer:
  - Is your company's product based on Superset in any way?
    - [ ] Y
    - [ ] N
  - Does your company use Superset internally?
    - [ ] Yes, we deploy it internally
    - [ ] Yes, through a hosted provider (e.g. Preset)
    - [ ] N
      - Why not?
        - Open answer (optional)

**Comment by rusackas** [comment]
Where do you turn for help with technical problems (Superset or otherwise)?
- [ ] Github
- [ ] Stack Overflow
- [ ] Slack
- [ ] Google
- [ ] Other (please specify)

**Comment by rusackas** [comment]
At the end of the survey, check all that apply:
-  I am open to receiving further updates/questions regarding this survey to help improve the product
  - [ ] Yes
  - [ ] No
- I am open to public disclosure that my organization uses Superset (y/n/I'll get back to you)
   - [ ] Yes
   - [ ] No
   - [ ] Maybe - I'll get back to you
I would like to join the Superset Community Newsletter
   - [ ] Yes
   - [ ] No
   - [ ] Maybe - I'll get back to you

**Reply by satokonak** [reply]
How should we use the second question? It's usually up to the PR team if they exist
